src/app.ts

```diff
@@ -168,7 +170,6 @@ async function app() {
     config = readConfig(originalInput);
     tank = fillTank(rawFSH, config);
   } catch {
-    program.outputHelp();
```

Comment:
Should we log _some_ message before we exit?  I agree that we shouldn't log `outputHelp`, but maybe a general message that we encountered an unexpected error?  Because I assume that if we make it to this catch block, it is an _unexpected_ error, right?  Or do we sometimes log a message and then throw intentionally?

Reply:
I'm a little torn.  On one hand, I don't love the idea of duplicate errors being logged.  On the other hand, I'm worried about an unexpected exception getting through and us not logging it at all (and just failing without error).

Would it be crazy to do something like this?
```ts
// If no errors have been logged yet, log this exception so the user knows why we're exiting
if (stats.numError === 0) {
  logger.error(`An unexpected error has occurred: ${e.message ?? e}`, e);
}
```

Reply:
Hm okay I added it in [dac7158](https://github.com/FHIR/sushi/pull/1102/commits/dac71586903a6e399649317797f445c448bea934) but I realized that this might not work in every scenario. I think there are places (like `findInputDir` and `ensureOutputDir`) where we log an error with `logger.error` but do not throw an error. That means in those places, SUSHI will carry on as normal, and if it does hit an unexpected error, it will go into this catch block but not log the "unexpected error" message. I think this approach might still be the most ideal approach we have, but I wanted to highlight that gap I realized.

Reply:
Yep! Sounds good.